### ISSUE TITLE
After updating to v0.3.1 alpha

### ISSUE THREAD

**Issue by SchRita** [TypesettingTools/ASSFoundation#2]
First of all, thank you for fixing the issue.

However, I still get the following errors when starting Aegisub:
http://pastebin.com/6gJeUN18
and when trying to convert a text to clip/drawing, I get the following:
http://pastebin.com/1tHuwHk4

A screenshot of my Aegisub:
![assf_issue_aegisub schrita](https://cloud.githubusercontent.com/assets/5681653/9815689/49f58d24-5898-11e5-8cc8-4c3e83b712cb.png)

And another of my Nudge options:
![assf_issue_nudge schrita](https://cloud.githubusercontent.com/assets/5681653/9815698/5d35c00c-5898-11e5-9b31-de1efd7171ef.png)

I might be doing something terribly stupid but I don't know what. :(
Sorry for taking up some of your time.

**Comment by torque** [comment]
Looks like one of the commits didn't quite fix everything regarding the `getTextMetrics` change.

[Section/Text.lua#L65-66](https://github.com/TypesettingTools/ASSFoundation/blob/master/l0/ASSFoundation/Section/Text.lua#L65-L66) I guess should be using `.bounds` instead of `.typeBounds`.

As for the loading error, something weird is happening. I suspect Yutils is somehow getting loaded twice, and its own `ffi.cdef` is conflicting with itself, though I'm not exactly sure what the scope on `ffi.cdef` is. As for how that would happen, I have absolutely no clue.

**Comment by SchRita** [comment]
Thanks. ^^ I'll reset my %AppData%/Aegisub, install DepCtrl and download Nudge to see if it works. I have all the available scripts downloaded but I don't know why that would cause this kind of error. If you need any more information that could be helpful, let me know.

**Comment by line0** [comment]
> Looks like one of the commits didn't quite fix everything regarding the getTextMetrics change.

Indeed. Should be all fine now.

> As for the loading error, something weird is happening. I suspect Yutils is somehow getting loaded twice, and its own ffi.cdef is conflicting with itself, though I'm not exactly sure what the scope on ffi.cdef is. As for how that would happen, I have absolutely no clue.

It happens because the previous versions of _VerticalText_ and _MoveAlongPath_ as well as _ASSFoundation_ loaded _Yutils_ as "YUtils" which was fixed in @5d0473d95498c82808e794c6ec7e930f0d931004 as well as [in the scripts](https://github.com/TypesettingTools/line0-Aegisub-Scripts/commit/c27a89a04c71388211891fceb5a458ccb75b69c9). 
While @SchRita got the _ASSFoundation_ update, it wasn't time yet for the _VerticalText_ and _MoveAlongPath_ update checks. So each script both loaded _**YU**tils_ while also loading _ASSFoundation_ which at that point loaded _**Yu**tils_ and since the `package.loaded` keys
are case-sensitive, _Yutils_ got loaded twice in the same Lua environment. The rest should be obvious.

tl;dr update those 2 scripts as well and everything should be fine.

**Comment by SchRita** [comment]
Thank you so much!
Text to **drawing** works awesomely. But when I tried the same with text to **clip** I get something similar than the issue before:
http://pastebin.com/5HvC4Vt8
I really feel obnoxious about reporting it, it may be a stupid thing on my end (although I tested it again and it just won't do anything else).
Thank you for your hard work!

**Comment by line0** [comment]
> Text to drawing works awesomely. But when I tried the same with text to clip I get something similar than the issue before:
> http://pastebin.com/5HvC4Vt8

This one's an unrelated problem, though. Can you please open a new issue and attach an .ass file with the line(s) you were trying Nudge on, including all required styles? I can't reproduce that one with simple test data. 

> I really feel obnoxious about reporting it, it may be a stupid thing on my end (although I tested it again and it just won't do anything else).

Even if that's the case, the script should still point it out rather than simply failing. By all means, keep these reports coming - so far quite a few of the ASSFoundation features appear to only have ever been tested by me, so you finding these regressions is a welcome change.
